lib/instrumentation/http-shared.js

```diff
@@ -206,6 +206,15 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
           status_code: statusCode,
           url
         })
+
+        if (typeof statusCode !== 'undefined') {
+          if (statusCode >= 400) {
+            span.outcome = 'failure'
+          } else {
+            span.outcome = 'success'
+          }
+        }
+
```

Comment:
Here I see we're setting the `outcome` property of a span based on the the HTTP status code -- anything greater than 400 is a `failure`. This matches the behavior described in the spec.  The inline type checking seems prudent given how the lifecycle of the `statusCode` variable is a little weird in this module.

Reply:
I've add a comment in `span.setOutcome` with regards to the type checking to make it clear